rfc/previews-using-fetaure-flags.md

```diff
@@ -0,0 +1,306 @@
+---
+title: 'Supergraph Previews for every Pull Request'
+author: Suvij Surya
+---
+
+This RFC proposes a way to create a Supergraph Preview for every pull request. This would allow developers to test their changes before merging them.
+This would be a github action managed by the WunderGraph Cosmo team.
+
+## Introduction
+
+When developing a new feature, it's important to test it in a production-like environment. By creating a preview for every pull request, developers can test their changes in a production-like environment before merging them.
+
+With [Graph Feature Flags](https://cosmo-docs.wundergraph.com/concepts/feature-flags), you can set up your continuous integration pipeline to deploy a new feature subgraph and feature flag for every pull request. This way, you can create one composition for every pull request without having to deploy a whole new set of subgraphs. All we need is a staging environment that's deployed from all main branches of your subgraphs. With graph feature flags, we can then "override" the subgraphs for a specific pull request.
+
+## Motivation
+
+Creating a Supergraph Preview for every pull request ensures that developers can thoroughly test their changes in an environment that closely mirrors production. This practice reduces the risk of introducing bugs or breaking changes when merging code, leading to more stable and reliable releases.
+
+## Proposal-1
+
+### Introduction
+
+This proposal creates and handles feature flags and feature subgraphs for each pull request using a predefined configuration file (cosmo.yaml). This approach uses the actions on the PR as triggers to create, update, and destroy the feature flags and feature subgraphs.
+
+```yaml
+# feature-flags-preview.yaml
+on:
+  pull_request:
+    paths:
+      - 'services/subgraph1/**/*.graphql'
+      - 'services/subgraph2/**/*.graphql'
+    types: [labeled, synchronize, closed, opened, reopened]
+
+jobs:
+  create:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.action == 'opened' || github.event.pull_request.action == 'reopened'
+    steps:
+      # 1. Create & Publish feature subgraph based on cosmo.yaml
+      # 2. Create a new feature flag
+      - uses: wundergraph/feature-flags@main
```

Comment:
do we call this "feature flags", "previews", or something more generic, like "ci" or "continous-integration-toolkit"

Reply:
Do you mean the name of the action?

Reply:
yes

Reply:
Yeah, the name can be anything we decide